test/e2e/e2e_test.go

```diff
@@ -1,97 +0,0 @@
-)
-
-const KIND = "kind"
-
-var _ = Describe("Workload cluster creation", func() {
-
-	BeforeEach(func() {
-		osType := strings.ToLower(os.Getenv("OS"))
-		Expect(osType).ToNot(Equal(""))
-		validateGlobals(specName)
-
-		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
-		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
-	})
-
-	AfterEach(func() {
-		dumpSpecResourcesAndCleanup(ctx, specName, bootstrapClusterProxy, artifactFolder, namespace, e2eConfig.GetIntervals, clusterName, clusterctlLogFolder, skipCleanup)
-	})
-
-	Context("Creating a highly available control-plane cluster", func() {
-		It("Should create a cluster with 3 control-plane and 1 worker nodes", func() {
-			By("Creating a high available cluster")
-			if upgradeTest {
-				upgradeManagementCluster()
-			} else {
-				controlPlaneMachineCount = int64(*e2eConfig.GetInt32PtrVariable("CONTROL_PLANE_MACHINE_COUNT"))
-				workerMachineCount = int64(*e2eConfig.GetInt32PtrVariable("WORKER_MACHINE_COUNT"))
-				result := &clusterctl.ApplyClusterTemplateAndWaitResult{}
-				clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
-					ClusterProxy: bootstrapClusterProxy,
-					ConfigCluster: clusterctl.ConfigClusterInput{
-						LogFolder:                clusterctlLogFolder,
-						ClusterctlConfigPath:     clusterctlConfigPath,
-						KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
-						InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
-						Flavor:                   osType,
-						Namespace:                namespace,
-						ClusterName:              clusterName,
-						KubernetesVersion:        e2eConfig.GetVariable(KubernetesVersion),
-						ControlPlaneMachineCount: &controlPlaneMachineCount,
-						WorkerMachineCount:       &workerMachineCount,
-					},
-					CNIManifestPath:              e2eConfig.GetVariable(capi_e2e.CNIPath),
-					WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
-					WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
-					WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
-				}, result)
-				cluster = result.Cluster
-				targetCluster = bootstrapClusterProxy.GetWorkloadCluster(ctx, namespace, clusterName)
-				remediation()
-				if !ephemeralTest {
-					pivoting()
-					upgradeBMO(targetCluster.GetClientSet())
-					upgradeIronic(targetCluster.GetClientSet())
-					certRotation(targetCluster.GetClientSet(), targetCluster.GetClient())
-					nodeReuse(targetCluster.GetClient())
-					rePivoting()
-				} else {
-					upgradeBMO(bootstrapClusterProxy.GetClientSet())
-					upgradeIronic(bootstrapClusterProxy.GetClientSet())
-					certRotation(bootstrapClusterProxy.GetClientSet(), bootstrapClusterProxy.GetClient())
-					nodeReuse(bootstrapClusterProxy.GetClient())
-
-				}
-			}
-		})
-	})
-})
```

Comment:
Dumping this file completely and renaming it to "features_test.go" does not look the right direction to take in order to put parallelism in place. I will list all the main concerns I have and we have discussed during the call offline:

- We are shifting from CAPI structured e2e test file names and conventions for file names across the providers. One can argue this, but I prefer sticking to common test file names for a better user experience
- In the current state, we are leaving remediation tests alone (by the way this test also incorporates inspection tests in the middle) outside and planning to have them separate for purposes of running them in parallel. However, that way we are kind of saying that remediation+inspection tests are not the features but whatever tests we have in a new features_test.go are. 
- How are you planning to keep this scalable, i,e in the future? **how does**/**based on what** the one would decide where to add/run the new feature e2e test (in the features_test.go OR separately as you have remediation test)?

Personally, I had imagined the parallelism as a bit different approach, where we could run all the tests in the same VM under the hood, in one go. Since that is not the case (we would have 2 separate VMs running different tests at the same time), I question myself what is the benefit of it over a new job/trigger you could introduce, which after all both of them would consume the same resources. 
If needed, we could classify and gate our current e2e tests, so we have different different triggers based on that classification, as [CAPI](https://github.com/kubernetes-sigs/cluster-api/pull/6411) or [CAPZ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2235#issuecomment-1101038608) for example does:

- test-v1b1-ubuntu-e2e-main
- test-v1b1-ubuntu-e2e-optional
- test-v1b1-ubuntu-capi-e2e